### ISSUE TITLE
Feature Request: Configure no container VPN and more

### ISSUE THREAD

**Issue by brochard** [mozilla/multi-account-containers#2247]
With MozillaVPN being integrated there are many good scenario that I would like to be able to make easily.
When I discovered it I wanted to have every big company have a different IP so I devided them in their container and linked them to MozillaVPN, but I think it could be made easier.

1) You should be able to configure the "no container" category
2) You should be able to configure so that every tab of a category have a different VPN connection.

Since Total Cookie Protection got added, making a container for every big tech is overkill but I would like to have a different IP, having every tab connect to a new IP would be amazing.

Tell me if I'm missing something, but organising things like that is not trivial and overkill in my opinion.
![image](https://user-images.githubusercontent.com/31168075/146847909-5b87f00b-d16c-4ac3-bbb6-6c680e95676f.png)

**Comment by dannycolin** [comment]
> You should be able to configure so that every tab of a category have a different VPN connection

Do you mean that every new tab from Container X would randomly get assigned to a different VPN server?

**Comment by brochard** [comment]
> Do you mean that every new tab from Container X would randomly get assigned to a different VPN server?

Yeah sorry for the bad github post, it's kind of a weird request but yeah that's the idea, being able to
1) configure the vpn connection of tabs not using containers
2) have the possibility to have random separate connection per tabs
The second idea might be crazy or impossible I truly don't know sorry but it could be used to improve privacy and to be way easier to configure than making different containers for amazon, google, facebook...
3) there could also be pre configured containers for those big tech companies that you could activate (right now making them is quite some work getting all their domains listed)

Thanks for answering this weird post ^^

**Comment by dannycolin** [comment]
> configure the vpn connection of tabs not using containers

This would need to be implemented directly in Firefox. Currently, you can configure a http(s)/socks proxy that applies by default to all tabs (in and out of a container) in about:preferences -> Network Settings. However to excludes containers from it, Firefox needs a way to communicate with Mozilla VPN and do some split-tunneling.

> there could also be pre configured containers for those big tech companies

I also thought about it myself. A curated list would indeed be very nice especially for less techy folks that would like a bit more privacy and/or want to try containers. Would you mind opening a different ticket for this idea?

> have the possibility to have random separate connection per tabs

It is probably doable for containerized tabs but would probably be complexe to implement it right. It'd need some good UX to make sure the user knows which VPN server is in use. There'll be an increased chance of hitting geofences (website blocking a geographic location). However, it's a nice idea. Do you mind if we limit this ticket to this idea?

**Comment by brochard** [comment]
> This would need to be implemented directly in Firefox.

From a technical perspective it might be logical to be implemented in the browser but I feel like it's more logical for a user when using Multi Account Container to configure VPNs connections to also be able to configure the connection of pages not containered.

> Would you mind opening a different ticket for this idea?

It looks like feature requests aren't supposed to be here to I made one on [Mozilla Connect](https://connect.mozilla.org/t5/ideas/multi-account-containers-pre-configured-containers-for-big-tech/idi-p/11525#M5737) (waiting for the post to be approved)


> Do you mind if we limit this ticket to this idea?

No problem


Since this ticket is already a mess I'll add that it would be nice to assign a bypass of vpn to a container (like you can exclude apps from the vpn apps), useful for sites banning vpn connections.
There's also the problem of not being able to access sites assigned to a container with vpn when your vpn is off. (there should maybe be an alert that you're trying to access the site without a vpn that could then be bypassed)
I'll probably make other Mozilla Connect suggestions, feel free to post them separatly on github if you want to clean this mess I made, sorry ^^.

**Comment by dannycolin** [comment]
> Since this ticket is already a mess I'll add that it would be nice to assign a bypass of vpn to a container (like you can exclude apps from the vpn apps), useful for sites banning vpn connections.

That'd be https://github.com/mozilla/multi-account-containers/issues/2259

> It looks like feature requests aren't supposed to be here to I made one on [Mozilla Connect](https://connect.mozilla.org/t5/ideas/multi-account-containers-pre-configured-containers-for-big-tech/idi-p/11525#M5737) (waiting for the post to be approved)

Thanks. Subscribed!

> From a technical perspective it might be logical to be implemented in the browser but I feel like it's more logical for a user when using Multi Account Container to configure VPNs connections to also be able to configure the connection of pages not containered.

Agree we should also be able to configure it through this addon. This would need to be exposed in the WebExtension API so again something that need to be done upstream :P.
